src/mpl/src/shm/mpl_shm_posix.c

```diff
@@ -66,6 +66,15 @@ void *MPL_initshm_open(const char *name, int size, bool * is_root_p)
         if (!fd) {
             goto fn_fail;
         }
+
+        /* wait until the region size is set before mapping */
+        struct stat statbuf;
+        do {
+            int rc = fstat(fd, &statbuf);
+            if (rc) {
+                goto fn_fail;
+            }
+        } while (statbuf.st_size != size);
```

Comment:
How about `while (statbuf.st_size == 0)' and add an assertion for checking whether the size is expected?

I worry a glitch may put it to deadlock

Reply:
updated